### PR TITLE
OJ-1960: Apply monorepo config from the templates repo

### PR DESCRIPTION
**Pull request by CharlesIC**
- Apply dotfiles from the repo template
- Run pre-commit
- Apply monorepo config to lambdas
  Update common config and resolve issues with some failing unit tests
- Apply monorepo config to integration tests
  Update common config for integration tests

**Apply monorepo config to the infrastructure temple**

- Apply changes from the lambda repo template
- Fix issues in the template, such as incorrect use of `!Ref` and `!GetAtt`
- Apply consistent formatting to the template to simplify it and make it easier to read
- Use the `LanguageExtensions` transform to use additional functionality, such as converting YAML object to JSON
- Use the `GetAtt` function to get the ARNs of the log groups
- Fix typo in `PersonalIdenityTable` output name
- Remove unused outputs
- Set the correct OpenAPI version in the resources - we use 3.0.1 (not 3.1.0, which is not supported by CF)

- Use references to the parameter resources to get their names
  If the name changes, it doesn't need to be updated in two places.

- Use `/aws/vendedlogs` as the prefix for all log groups
  This helps to avoid the account log delivery service policy size breach issue

- Remove private API duplication from the SAM template
  The private and "dev only" APIs are the same except for two properties - use the dev env condition to set the correct values for those properties instead of duplicating the whole API block and using complex logic to select the correct ARN for the output.

  - Delete the "dev only" API as it is just the private API in dev mode (no auth and type is regional instead of private)

  - Use the `IsDevEnvironment` condition, and remove `IsNotDevEnvironment` since it's not necessary

### CODE REVIEW

**Comment by jkunle** [comment]
When I tried to build this I get 
```
 AWS_PROFILE=cri-kbv-hmrc-dev sam build
Building codeuri: /Documents/di-ipv-cri-kbv-hmrc-api/lambdas/submit-answer runtime: nodejs18.x metadata: {'BuildMethod': 'esbuild', 'BuildProperties': {'Minify': True,
'Target': 'es2020', 'Sourcemap': True, 'EntryPoints': ['src/submit-answer-handler.ts']}} architecture: arm64 functions: SubmitAnswerFunction                                                 
 Running NodejsNpmEsbuildBuilder:CopySource                                                                                                                                                  
 Running NodejsNpmEsbuildBuilder:NpmInstall                                                                                                                                                  
 Running NodejsNpmEsbuildBuilder:EsbuildBundle                                                                                                                                               

Build Failed
Error: NodejsNpmEsbuildBuilder:EsbuildBundle - Esbuild Failed: Cannot find esbuild. esbuild must be installed on the host machine to use this feature. It is recommended to be installed on the PATH, but can also be included as a project dependency.
```

**Review by surajkumar — APPROVED** [review]

File: deploy.sh [hunk 1]
```diff
@@ -1,29 +1,29 @@
 #!/usr/bin/env bash
-set -e
+set -eu
 
-stack_name="$1"
-common_stack_name="${2:-kbv-hmrc-common-cri-api-local}"
```
Comment (jkunle) [inline deploy.sh:5]: Can we keep this, need to able to specify a different common-stack, devs can deploy a common-stack different from `common-cri-api`

Comment (CharlesIC) [reply deploy.sh:5]: Yes but the default needs to be empty then as the value in the script is incorrect so local deployments fail. The script on main doesn't work - this makes it work with the default common stack.

Comment (jkunle) [reply deploy.sh:5]: And then deploy their dev stack to point to it 

`./deploy.sh ola-kbv-hmrc-cri ola-common-cri-api`

Comment (CharlesIC) [reply deploy.sh:5]: OK do you mind if I do that in the next PR?

Comment (CharlesIC) [reply deploy.sh:5]: Looks like that stack doesn't exist anymore so we can make the default empty and then the `common-cri-api ` will be used, or you'll be able to run the command you posted above to override it. We just don't want a bad default in the script

**Pull request merged**